apps/cowswap-frontend/src/modules/permit/utils/handlePermit.ts

```diff
@@ -30,9 +30,7 @@ export async function handlePermit(params: HandlePermitParams): Promise<AppDataI
       throw new Error(`Unable to generate permit data`)
     }
 
-    const hooks = buildAppDataHooks({
-      preInteractionHooks: [permitData],
-    })
+    const hooks = addPermitHookToHooks(typedHooks, permitData)
```

Comment:
Here is where the "merge" happens. 
Permit is not added to the order anywhere else.